projects/plutus/src/ltc2484.c

```diff
@@ -0,0 +1,132 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "ltc2484.h"
+
+uint8_t s_filter_modes[NUM_LTC_2484_FILTER_MODES] = {
+  LTC2484_REJECTION_50HZ_60HZ,
+  LTC2484_REJECTION_50HZ,
+  LTC2484_REJECTION_60HZ,
+};
+
+StatusCode ltc2484_init(const Ltc2484Settings *config) {
+  if (config->filter_mode >= NUM_LTC_2484_FILTER_MODES) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+
+  // The LTC2484 uses SPI Mode 0 (see Figure 5 on p.20 in the datasheet)
+  SPISettings spi_config = {
+    .baudrate = config->spi_baudrate,
+    .mode = SPI_MODE_0,
+    .mosi = config->mosi,
+    .miso = config->miso,
+    .sclk = config->sclk,
+    .cs = config->cs,
+  };
+
+  spi_init(config->spi_port, &spi_config);
+
+  uint8_t data[1] = { LTC2484_ENABLE | LTC2484_EXTERNAL_INPUT | LTC2484_AUTO_CALIBRATION |
+                      s_filter_modes[config->filter_mode] };
+
+  return spi_exchange(config->spi_port, data, 1, NULL, 0);
```

Comment:
Don't we still need to clock all 32 bits?

Reply:
tbh, I'm not sure. The datasheet isn't too clear about that:

>  These special features are selected through a single 8-bit serial input word during the data input/output cycle (see Figure 2).
> ...
> In order to change the mode of operation, an enable bit (EN) followed by up to 7 bits of data are shifted into the device (see Table 1). 

I'll change this back to 4 bytes just to be safe :)

Reply:
Cool, we should probably test it before we merge just to be sure.